woltka/file.py

```diff
@@ -385,6 +392,7 @@ def write_table(fh, data, samples=None, tree=None, rankdic=None, namedic=None,
     print('\t'.join(header), file=fh)
 
     # table body
+    nrow = 0
     for key in sorted(allkeys(data)):
```

Comment:
would enumerate have the same effect than nrow?

Reply:
Good point! I tested a bit. The goal for this `nrow` is to calculate how many keys are there in total. If I do `for i, key in enumerate(...):`, this `i` variable may become undefined if there is no key in the dict, which is a normal case rather than an error because it is completely possible that the classification process returns no result. To avoid this error, I would recommend the original form.